File: JavaScript/7js 2/lateRide.js
/*
|--------------------------------------------------------------------------
| Simple Fun #3: Late Ride
|--------------------------------------------------------------------------
|
*/

function lateRide(n) {
   const arr = String(n % 60).split('').concat(String(Math.floor(n / 60)).split(''))
   return arr.map(i => Number(i)).reduce((a, b) => a + b);
}


function lateRide(n) {
   return [...`${n / 60 ^ 0}` + n % 60].reduce((pre, val) => pre + +val, 0);
}


function lateRide(n) {
   return [~~(n / 60), n % 60].join('').split('').reduce((a, b) => a + +b, 0);
}


console.log(lateRide(240), 4)
console.log(lateRide(808), 14)
console.log(lateRide(1439), 19)

File: Python/7py 2/late_ride.py
"""
|--------------------------------------------------------------------------
| Simple Fun #3: Late Ride
|--------------------------------------------------------------------------
|
"""

def late_ride(n):
   return sum(list(map(lambda i: int(i), list(str(n//60))+list(str(n%60)))))


def late_ride(n):
   return sum(map(int, str(n//60)+str(n%60)))


def late_ride(n):
   return sum([int(x) for x in (str(n % 60) + str(int(n / 60)))])


print(late_ride(240), 4)
print(late_ride(808), 14)
print(late_ride(1439), 19)
print(late_ride(0), 0)
print(late_ride(23), 5)
print(late_ride(8), 8)